tests/jax/core_system_components/launcher_test.py

```diff
@@ -15,4 +15,381 @@
+        == mock_parameter_server_fn
+    )
+    assert [parameter_server] == launcher._program._groups["parameter_server_test"][
+        -1
+    ]._created_handles
+
+    assert not hasattr(launcher, "_nodes")
+    assert not hasattr(launcher, "_node_dict")
+
+
+def test_add_multi_process_two_add_same_name(
+    mock_parameter_server_fn: Callable, mock_parameter_server_second_fn: Callable
+) -> None:
+    """Test calling twice add for two nodes with same name for the case of multi process
+
+    Args:
+        mock_parameter_server_fn
+        mock_parameter_server_second_fn
+    """
+    launcher = Launcher(multi_process=True)
+
+    parameter_server_1 = launcher.add(
+        mock_parameter_server_fn,
+        node_type=NodeType.corrier,
+        name="parameter_server_test",
+    )
+    parameter_server_2 = launcher.add(
+        mock_parameter_server_second_fn,
+        node_type=NodeType.corrier,
+        name="parameter_server_test",
+    )
+
+    assert list(launcher._program._groups.keys()) == ["parameter_server_test"]
+    assert (
+        launcher._program._groups["parameter_server_test"][0]._constructor
+        == mock_parameter_server_fn
+    )
+    assert (
+        launcher._program._groups["parameter_server_test"][1]._constructor
+        == mock_parameter_server_second_fn
+    )
+    assert [parameter_server_1] == launcher._program._groups["parameter_server_test"][
+        0
+    ]._created_handles
+    assert [parameter_server_2] == launcher._program._groups["parameter_server_test"][
+        1
+    ]._created_handles
+
+    assert not hasattr(launcher, "_nodes")
+    assert not hasattr(launcher, "_node_dict")
+
+
+def test_add_non_multi_process_reverb_node(mock_data_server_fn: Callable) -> None:
+    """Test the add method for the case of one process and for node_type reverb
+
+    Args:
+        mock_data_server_fn
+    """
+    launcher = Launcher(multi_process=False)
+    with pytest.raises(Exception):
+        data_server = launcher.add(
+            mock_data_server_fn,
+            node_type=NodeType.reverb,
+            name="data_server_test",
+        )
+    data_server = launcher.add(
+        mock_data_server_fn,
+        node_type=NodeType.reverb,
+        name="data_server",
+    )
+
+    assert not hasattr(launcher, "_program")
+
+    assert launcher._replay_server._port != None
+    assert type(launcher._replay_server._port) == int
+    assert isinstance(launcher._replay_server._server, pybind.Server)
+
+    assert launcher._nodes[-1] == launcher._node_dict["data_server"]
+    assert data_server == launcher._nodes[-1]
+
+    assert isinstance(launcher._node_dict["data_server"], Client)
+    assert launcher._node_dict["data_server"]._server_address == (
+        f"localhost:{launcher._replay_server._port}"
+    )
+    assert isinstance(launcher._node_dict["data_server"]._client, pybind.Client)
+    assert launcher._node_dict["data_server"]._signature_cache == {}
```

Comment:
You've already tested this above, you can remove it from here.

Reply:
it is the case of non-multi-process and with nodes of type reverb_node so we need to make a test since it is a different case.
What do you think?

Reply:
Ok sure, feel free to keep it then